vega_sim/scenario/common/agents.py

```diff
@@ -1473,6 +1475,301 @@ def _calculate_price_volume_levels(
+            fee_amount=fee_amount,
+            kappa=kappa,
+            num_levels=num_levels,
+            tick_spacing=tick_spacing,
+            inventory_upper_boundary=inventory_upper_boundary,
+            inventory_lower_boundary=inventory_lower_boundary,
+            terminal_penalty_parameter=terminal_penalty_parameter,
+            running_penalty_parameter=running_penalty_parameter,
+            market_order_arrival_rate=market_order_arrival_rate,
+            market_kappa=market_kappa,
+            asset_decimal_places=asset_decimal_places,
+            tag=tag,
+            key_name=key_name,
+            orders_from_stream=orders_from_stream,
+            state_update_freq=state_update_freq,
+        )
+
+        self.profit_margin = profit_margin
+
+        self.external_market_name = external_market_name
+        self.external_market_id = None
+
+        self.external_key_name = external_key_name
+        self.internal_delay = internal_delay
+        self.external_delay = external_delay
+
+        self.transfer_threshold = transfer_threshold
+
+    def initialise(
+        self,
+        vega: Union[VegaServiceNull, VegaServiceNetwork],
+        create_wallet: bool = True,
+        mint_wallet: bool = True,
+    ):
+        # Initialise the parent ExponentialShapedMarketMaker
+        super().initialise(vega, create_wallet, mint_wallet)
+
+        self.external_market_id = self.vega.find_market_id(
+            name=self.external_market_name
+        )
+        self._update_state(current_step=self.current_step)
+
+        if vega.create_wallet:
+            vega.create_wallet(
+                name=self.wallet_name,
+                passphrase=self.wallet_pass,
+                key_name=self.external_key_name,
+            )
+        if mint_wallet:
+            vega.mint(
+                wallet_name=self.wallet_name,
+                asset=self.asset_id,
+                amount=self.initial_asset_mint,
+                key_name=self.external_key_name,
+            )
+
+    def _update_state(self, current_step: int):
+        super()._update_state(current_step)
+
+        if self.state_update_freq and current_step % self.state_update_freq == 0:
+
+            int_market_info = (
+                self.vega.market_info(market_id=self.market_id)
+                if self.market_id is not None
+                else None
+            )
+            self.int_mkr_fee = (
+                float(int_market_info.fees.factors.maker_fee)
+                if int_market_info is not None
+                else 0
+            )
+            self.int_liq_fee = (
+                float(int_market_info.fees.factors.liquidity_fee)
+                if int_market_info is not None
+                else 0
+            )
+            self.int_inf_fee = (
+                float(int_market_info.fees.factors.infrastructure_fee)
+                if int_market_info is not None
+                else 0
+            )
```

Comment:
Could we collapse all of these into a single `if int_market_info is not None:` block? And is there ever a time where `self.market_id` is not None and `int_market_info` still is? If not, lets roll all four into one